tests/unit/test_report_junit.py

```diff
@@ -39,10 +42,78 @@ def get(key, default=None):
     return report, results, out_file_path
 
 
-def do_assert(out_file_path, expected):
-    with open(out_file_path) as f:
-        output = f.read().replace('\t', ' ' * 4)
-    assert output == expected
+# A simple way would be comparing XML nodes as strings, e.g. output of their
+# .toprettyxml() methods. But that has been observed as not fully deterministic
+# with Python 3.6, changing order or attributes from time to time. Therefore
+# taking the longer, more verbose approach.
+def _compare_xml_node(tree_path: List[str], expected: xml.dom.Node, actual: xml.dom.Node) -> None:
+    """ Assert two XML nodes have the same content """
+
+    # All of this would be doable in a much, much simpler, condensed manner,
+    # but being more verbose allows for way more specific error message when
+    # comparison fails.
+
+    tree_path_joined = '.'.join(tree_path)
+
+    # Make sure node names do match.
+    assert expected.nodeName == actual.nodeName, \
+        f"Element name mismatch at {tree_path_joined}: " \
+        f"expected {expected.nodeName}, " \
+        f"found {actual.nodeName}"
+
+    # If nodes have the same tag, move on to attributes. Make sure both nodes
+    # have the same set of attributes, with same respective values.
+    #
+    # Note: sometimes, node.attributes may be set to `None`, meaning "no attributes",
+    # which is fine as long as the other node also has no attributes.
+    expected_attributes = sorted((expected.attributes or {}).items())
+    actual_attributes = sorted((actual.attributes or {}).items())
+
+    for (expected_name, expected_value), (actual_name, actual_value) in zip(
```

Comment:
PASS even when number of attributes doesn't match (IMO we need explicit assert on count of attributes)).

E.g. this passes
```
  a = "<foo a='a'/>"
  b = "<foo a='a' b='b'/>"
  with xml.dom.minidom.parseString(a) as dom_a:
    with xml.dom.minidom.parseString(b) as dom_b:
      assert _compare_xml_node([dom_a.nodeName], dom_a, dom_b)
```

Reply:
Damn you, `zip`! Right, it's default behavior. Nice catch. And since we already have the same assert for the number of children, added one for attributes as well.